src/main/java/com/systemmeltdown/meltdownlog/topics/DoubleTopic.java

```diff
@@ -0,0 +1,29 @@
+package com.systemmeltdown.meltdownlog.topics;
+
+import com.systemmeltdown.meltdownlog.lib.Utils;
+
+public class DoubleTopic extends Topic {
+	private double m_roundingFactor;
+	private double m_lastRoundedValue = Double.MIN_VALUE;
+	private long m_lastNanos = Long.MIN_VALUE;
+
+	public DoubleTopic(String name, double roundingFactor) {
+		super(name, Double.TYPE);
+		m_roundingFactor = roundingFactor;
+	}
+
+	public void log(double value) {
+		double roundedValue = Utils.roundByFactor(value, m_roundingFactor);
+		if (roundedValue != m_lastRoundedValue) {
+			// Write the previous value first so we can get an accurate transition.
+			if (m_lastNanos != Long.MIN_VALUE) {
+				writeEntry(m_lastRoundedValue, m_lastNanos);
+			}
+
+			writeEntry(roundedValue);
+
+			m_lastRoundedValue = roundedValue;
+			m_lastNanos = System.nanoTime();
```

Comment:
It might be better to save this first and pass it to writeEntry, otherwise it's not the actual log time. Or return the log time from writeEntry.